packages/flutter/test/widgets/editable_text_test.dart

```diff
@@ -2846,6 +2846,116 @@ void main() {
+        focusNode: focusNode,
+        style: textStyle,
+        cursorColor: cursorColor,
+      ),
+    ));
+
+    final String expectedValue = '•' *controller.text.length;
+
+    expect(
+      semantics,
+      hasSemantics(
+        TestSemantics(
+          children: <TestSemantics>[
+            TestSemantics.rootChild(
+              children: <TestSemantics>[
+                TestSemantics(
+                  children: <TestSemantics>[
+                    TestSemantics(
+                      flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
+                      children: <TestSemantics>[
+                        TestSemantics(
+                          flags: <SemanticsFlag>[
+                            SemanticsFlag.isTextField,
+                            SemanticsFlag.isObscured,
+                          ],
+                          value: expectedValue,
+                          textDirection: TextDirection.ltr,
+                        ),
+                      ],
+                    )
+                  ],
+                )
+              ],
+            ),
+          ],
+        ),
+        ignoreTransform: true,
+        ignoreRect: true,
+        ignoreId: true,
+      ),
+    );
+
+    semantics.dispose();
+  });
+
+  testWidgets('password fields delayed hide the last char when obscure is ObscureTextBehavior.delayed', (WidgetTester tester) async {
+    final SemanticsTester semantics = SemanticsTester(tester);
+
+    controller.text = 'super-secret-password!!1';
+
+    await tester.pumpWidget(MaterialApp(
+      home: EditableText(
+        backgroundCursorColor: Colors.grey,
+        obscureTextBehavior: ObscureTextBehavior.delayed,
+        controller: controller,
+        focusNode: focusNode,
+        style: textStyle,
+        cursorColor: cursorColor,
+      ),
+    ));
+
+    final String expectedValue = '•' *controller.text.length;
+
+    expect(
+      semantics,
+      hasSemantics(
+        TestSemantics(
+          children: <TestSemantics>[
+            TestSemantics.rootChild(
+              children: <TestSemantics>[
+                TestSemantics(
+                  children: <TestSemantics>[
+                    TestSemantics(
+                      flags: <SemanticsFlag>[SemanticsFlag.scopesRoute],
+                      children: <TestSemantics>[
+                        TestSemantics(
+                          flags: <SemanticsFlag>[
+                            SemanticsFlag.isTextField,
+                            SemanticsFlag.isObscured,
+                          ],
+                          value: expectedValue,
+                          textDirection: TextDirection.ltr,
+                        ),
+                      ],
+                    )
+                  ],
+                )
+              ],
+            ),
+          ],
+        ),
+        ignoreTransform: true,
+        ignoreRect: true,
+        ignoreId: true,
+      ),
+    );
+
+    semantics.dispose();
+  });
+
```

Comment:
Could you also test that the text in the field is immediately changed to the obscuring character when set to `all`?  Something like this but without obscuringTextCharacter: https://github.com/flutter/flutter/blob/5acf7e98f16222610a4e6bfe1e36b08a90cf53a4/packages/flutter/test/widgets/editable_text_test.dart#L2961